R/utils/sensistivity_analysis_utils.R

```diff
@@ -0,0 +1,305 @@
+    rownames(row_top)[i] <- t
+  }
+  
+  
+  table <-
+    kbl(SA_info[, !colnames(SA_info) %in% c("Topic", "Nam", "ID", "Object")],
+        caption = "Sensitivity analyses summary.", booktabs = T, align = "c", digits = 3) %>%
+    kable_styling(
+      full_width = FALSE,
+      font_size = 20,
+      position = "left",
+      latex_options = c("repeat_header",
+                        "striped",
+                        "hold_position")
+    ) %>% landscape()
+  
+  for (i in 1:length(Topic$Topic)) {
+    Nam <- Topic$Nam[i]
+    item <- Topic$Topic[i]
+    if(item %in% SA_info$Topic)
+      eval(parse(
+        text = paste0(
+          "table <- table %>% pack_rows('",
+          Nam,
+          "', row_top['",
+          item,
+          "',1], row_top['",
+          item,
+          "', 2])"
+        )
+      ))
+  }
+  
+  SA_info <- SA_info[, !colnames(SA_info) %in% c("Nam", "ID")]
+  rownames(SA_info) <- NULL
+  kableExtra::save_kable(table, 
+                         file = file.path(dir_SensAnal, "Sensitivity_analysis.pdf", fsep = fsep))
+  return(SA_info)
+}
+
+
+
+#' 2. Function to start a new sensitivity analysis ----
+#' ----------------------------------------------------------- #
+#' 
+#' @param topic (character string)- Indicates the main topic of the sensitivity 
+#' analysis. This HAS to be either "landings", "discards", "surveys",
+#' "biological_Info", or "model".
+#' @param object (character string)- Indicates a specific item of 
+#' the main topic. For example, "Length_Comp", "Age_comp", "Growth", "Fecundity",...
+#' @param author (character string)- Who leads the sensitivity analysis and runs
+#' the model.
+#' @param folder_name (character string)- name of the folder to create for that 
+#' particular sensitivity analysis. This folder will be housed in the root folder 
+#' 'Sensitivity_Anal' and will hold the scipts  used to set up the model and both 
+#' models input and output files for this analysis.
+#' @param script_model (character string)- Name of the script used to create the 
+#' model.
+#' @param script_results (character string)- Name of the script used to analyze 
+#' the results from that sensitivty analysis.
+#' @param base_model (character string)- which model has been used as the basis 
+#' of this analysis (before any modification).
+#' @param new_model (character string)- name of the model you're gonna develop for
+#' this analysis.
+#' 
+#' @details
+#' The function creates:
+#' 1. one folder named as \code{folder_name} in the 
+#'    "shortspine_thornyhead_2023/model/Sensitivity_Anal" repertory. This folder will
+#'    house model input and output files,
+#' 2. the "Sensitivity_Analysis_Features.txt" file (in the previously created folder).
+#'    This file summaryzes the features of the sensitivity analysis 
+#'    (\code{topic}, \code{object}, \code{author}, ...)
+#' 3. Update the "Sensitivity_analysis.pdf" file housed in the root folder.
+#
+
+NewSensAnal <- function(topic = NULL,
```

Comment:
Can we add some error checking to this function so that if a sensitivity analysis with the same properties has already been created (ie. it exists in the sensitivities data frame AND the model directory is present), the user is notified and nothing happens?

I think this would be useful so that all sensitivity scripts can start with a call to this function to setup the sensitivity, and then use all of the necessary R code to modify the relevant SS data and control files.

Reply:
Thse checking error are  in the "Check the input arguments" section of the function I think. If this kind of things happen, the fuction stops and the user gets a notification about what is wrong (e.g. the model directory already exists). However, I'll add some checks (and options) so we can set up directories for diffderent models in the same sensitivity analysis.

I agree with you (I thik that was the idea), it would be great to start all sensitivity by calling this function and then build the R code to modify the model input files.

Reply:
You're right, missed that check somehow when I was reviewing it.